web/src/stream_list_sort.ts

```diff
@@ -72,12 +75,22 @@ export function has_recent_activity(sub) {
     return stream_topic_history.stream_has_topics(sub.stream_id) || sub.newly_subscribed;
 }
 
-export function has_recent_activity_but_muted(sub) {
+export function has_recent_activity_but_muted(sub: StreamSubscription): boolean {
     return has_recent_activity(sub) && sub.is_muted;
 }
 
-export function sort_groups(stream_ids, search_term) {
-    const stream_id_to_name = (stream_id) => sub_store.get(stream_id).name;
+export function sort_groups(
+    stream_ids: number[],
+    search_term: string,
+): {
+    same_as_before: boolean;
+    pinned_streams: number[];
+    normal_streams: number[];
+    dormant_streams: number[];
+    muted_pinned_streams: number[];
+    muted_active_streams: number[];
+} {
```

Comment:
It's probably better to make this return value have a named type, `StreamListSortResult`, that can be used by the caller; one rarely wants to define a type in a function definition.

Reply:
Ah yeah that makes sense, updated with that